### PR TITLE
Create a photo cache to speedup the contactsmenu

### LINKED ISSUE #4653: Images of profile pictures in VCF should be compressed by the server
For example the picture of Björn in the contacts menu is 1 MB big… – We should add caching (https://github.com/nextcloud/server/issues/4637) but also compress the images on the server.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #4653 #4639 

* Creates a photo cache so we don't have to parse the vcard all the time.
* Stores resized versions so we don't have to resize
* Request sane size in the contacts menu
  * with srcset for hidpi screens

Todo:
- [x] Fix tests

### CODE REVIEW

File: apps/dav/lib/CardDAV/ImageExportPlugin.php [hunk 1]
```diff
@@ -85,90 +96,28 @@ function httpGet(RequestInterface $request, ResponseInterface $response) {
...
-			$response->setStatus(200);
+		$response->setHeader('Cache-Control', 'private, max-age=3600, must-revalidate');
+		$response->setHeader('Etag', $node->getETag() );
+		$response->setHeader('Pragma', 'public');
 
-			$response->setBody($result['body']);
+		try {
+			$file = $this->cache->get($addressbook->getResourceId(), $node->getName(), $size, $node);
+			$response->setHeader('Content-Type', $file->getMimeType());
+			$response->setHeader('Content-Disposition', 'inline');
```
Comment (LukasReschke) [inline apps/dav/lib/CardDAV/ImageExportPlugin.php]: attachment? 😄

**Comment by rullzer** [comment]
We should probabaly add more tests. But I also want this in ASAP. So lets add more tests later.

**Review by ChristophWurst — APPROVED** [review]
Tested and works nicely on both FF and Chrome 👍

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by MorrisJobke** [comment]
The install tests somehow fail, but they work locally. I will merge this here and have a look at how the merge CI job looks like.

**Comment by MorrisJobke** [comment]
> The install tests somehow fail, but they work locally. I will merge this here and have a look at how the merge CI job looks like.

Ran fine there: https://drone.nextcloud.com/nextcloud/server/7856/26

**Pull request merged**